apps/platform/src/app/(main)/(project)/[workspace]/[project]/@secret/page.tsx

```diff
@@ -40,16 +41,19 @@ function SecretPage(): React.JSX.Element {
   const selectedSecret = useAtomValue(selectedSecretAtom)
   const [secrets, setSecrets] = useAtom(secretsOfProjectAtom)
   const selectedProject = useAtomValue(selectedProjectAtom)
+  const isDecrypted = useAtomValue(shouldRevealSecretEnabled)
 
   const [page, setPage] = useState(0)
   const [hasMore, setHasMore] = useState(true)
   const [isLoading, setIsLoading] = useState<boolean>(true)
 
-  const isDecrypted = useMemo(
-    () => selectedProject?.storePrivateKey === true || false,
+  const privateKey = useMemo(
+    () =>
+      selectedProject?.storePrivateKey
+        ? selectedProject.privateKey
+        : localStorage.getItem(`${selectedProject?.name}_pk`) || '',
```

Comment:
we should return `null` instead of ''. This would help us to catch unwanted bugs by checkign for the logic. also, make `privateKey` of type `string | null`

Reply:
the related logic should also be implemented in SecretCard